Libs/MRML/Core/vtkMRMLMessageCollection.h

```diff
@@ -0,0 +1,98 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#ifndef __vtkMRMLMessageCollection_h
+#define __vtkMRMLMessageCollection_h
+
+// MRML includes
+#include "vtkMRML.h"
+
+// VTK includes
+#include <vtkIdTypeArray.h>
+
+// STD includes
+#include <string>
+#include <vector>
+
+///
+/// A class for recording warnings and errors associated with this
+/// vtkMRMLStorableNode.  A substantially similar vtkMessageCollection
+/// is planned for vtkAddon.  When that is released, this will be
+/// replaced by that.
+class VTK_MRML_EXPORT vtkMRMLMessageCollection : public vtkObject
+{
+public:
+  vtkTypeMacro(vtkMRMLMessageCollection,vtkObject);
+  void PrintSelf(ostream& os, vtkIndent indent) override;
+  static vtkMRMLMessageCollection* New();
+
+  ///
+  /// A helper class that describes a single message
+  struct Message
+  {
+    Message(unsigned long messageType, const std::string &messageText);
+    ///
+    /// Function object for checking whether a Message has a specified
+    /// MessageType.
+    struct EqualMessageType
+    {
+      EqualMessageType(unsigned long messageType);
+      bool operator()(const Message &message) const;
+
+      unsigned long MessageType;
+    };
+
+    unsigned long MessageType;
+    std::string MessageText;
+  };
+
+  ///
+  /// Return the total number of reported messages.
+  virtual int GetNumberOfMessages() const;
+
+  ///
+  /// Return the number of reported messages of the specified event
+  /// type
+  virtual int GetNumberOfMessagesOfType(unsigned long messageType) const;
+
+  ///
+  /// Return the number of reported messages of the specified event
+  /// type
+  virtual int GetNumberOfMessagesOfType(const char *eventName) const;
+
+  // Get the nth message from the message vector
+  virtual Message GetMessage(int index) const;
```

Comment:
"Message" type is not Python-wrapped. Add GetNthMessageText and GetNthMessageType (or just GetMessageText, GetMessageType) methods to make this class usable from Python.

Reply:
I replaced `GetMessage(int)` with `GetNthMessageType(int)` and `GetNthMessageText(int)`.

Reply:
And I moved the `struct vtkMRMLMessageCollection::Message` from public to protected.